pkg/providers/vault/secret_manager.go

```diff
@@ -33,13 +36,20 @@ type HttpSecretStoreManager struct {
 func NewSecretClient(config SecretConfig) (client pkg.SecretClient, err error) {
 	switch config.Provider {
 	case HTTPProvider:
+		httpClient, cErr := createHttpClient(config)
+		if cErr != nil {
+			err = cErr
+			return
+		}
```

Comment:
You can just use `err` here like so:
```
httpClient, err := createHttpClient(config)
if err != nil {
	return
}
```
I also notice the return signature on this function is somewhat unique from the rest of the code in that it defines the names of its return vars rather than just their types. Would recommend this not be the case for this PR as they should all do this or not. If you'd like this stylistically then create an issue to do it for all functions in the module and handle that effort separately.